### PR TITLE
Unlock with fee cleanup

### PR DESCRIPTION
**Pull request by dorin-iancu**
- use configurable values for penalty calculation instead of hard-coded ones. `lock options` are no longer just an u64, but a struct. Each entry keeps its lock option and the penalty percentage from which it starts.
- cleanup token merging logic
- move burning and sending to fees collector logic to a separate module

### CODE REVIEW

File: locked-asset/simple-lock-energy/src/unlock_with_penalty.rs [hunk 1]
```diff
@@ -227,135 +176,20 @@ pub trait UnlockWithPenaltyModule:
...
-                self.fees_from_penalty_unlocking()
-                    .set(NonceAmountPair::new(token_nonce, remaining_amount));
-            } else {
-                self.merge_fees_from_penalty(token_nonce, remaining_amount)
-            }
-        }
-
-        // Only once per week
-        self.send_fees_to_collector();
+        token_amount * penalty_percentage_unlock / MAX_PENALTY_PERCENTAGE as u64
```
Comment (sasurobert) [inline locked-asset/simple-lock-energy/src/unlock_with_penalty.rs:193]: clippy.

File: locked-asset/simple-lock-energy/src/unlock_with_penalty.rs [hunk 2]
```diff
@@ -199,22 +155,15 @@ pub trait UnlockWithPenaltyModule:
...
+        let penalty_percentage_full_unlock_current_epoch =
+            self.calculate_penalty_percentage_full_unlock(lock_epochs_remaining);
+        let penalty_percentage_full_unlock_target_epoch =
+            self.calculate_penalty_percentage_full_unlock(target_epoch_to_reduce);
 
         (penalty_percentage_full_unlock_current_epoch - penalty_percentage_full_unlock_target_epoch)
-            * MAX_PERCENTAGE as u64
-            / (MAX_PERCENTAGE as u64 - penalty_percentage_full_unlock_target_epoch)
+            * MAX_PENALTY_PERCENTAGE as u64
+            / (MAX_PENALTY_PERCENTAGE as u64 - penalty_percentage_full_unlock_target_epoch)
```
Comment (sasurobert) [inline locked-asset/simple-lock-energy/src/unlock_with_penalty.rs:166]: clippy

File: locked-asset/simple-lock-energy/src/unlock_with_penalty.rs [hunk 3]
```diff
@@ -199,22 +155,15 @@ pub trait UnlockWithPenaltyModule:
...
-            );
+        let penalty_percentage_full_unlock_current_epoch =
+            self.calculate_penalty_percentage_full_unlock(lock_epochs_remaining);
+        let penalty_percentage_full_unlock_target_epoch =
+            self.calculate_penalty_percentage_full_unlock(target_epoch_to_reduce);
 
         (penalty_percentage_full_unlock_current_epoch - penalty_percentage_full_unlock_target_epoch)
-            * MAX_PERCENTAGE as u64
-            / (MAX_PERCENTAGE as u64 - penalty_percentage_full_unlock_target_epoch)
+            * MAX_PENALTY_PERCENTAGE as u64
```
Comment (sasurobert) [inline locked-asset/simple-lock-energy/src/unlock_with_penalty.rs:165]: clippy

File: locked-asset/simple-lock-energy/src/fees.rs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    + elrond_wasm_modules::pause::PauseModule
+    + crate::penalty::LocalPenaltyModule
+    + crate::energy::EnergyModule
+    + crate::events::EventsModule
+    + crate::lock_options::LockOptionsModule
+    + utils::UtilsModule
+{
+    fn burn_penalty(&self, token_id: TokenIdentifier, token_nonce: Nonce, fees_amount: &BigUint) {
+        let fees_burn_percentage = self.fees_burn_percentage().get();
+        let burn_amount = fees_amount * fees_burn_percentage as u64 / MAX_PENALTY_PERCENTAGE as u64;
```
Comment (sasurobert) [inline locked-asset/simple-lock-energy/src/fees.rs:36]: clippy

**Review by CostinCarabas — APPROVED** [review]

**Comment by CostinCarabas** [comment]
Looks good to me.

**Review by psorinionut — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**